composer.json

```diff
@@ -62,10 +62,11 @@
         "mikey179/vfsstream": "^1.6.10",
         "mockery/mockery": "^1.6",
         "paragonie/csp-builder": "^2.3 || ^3.0",
-        "phpstan/phpstan": "1.12.0",
         "phpstan/extension-installer": "^1.3",
-        "symplify/phpstan-rules": "^12.4",
-        "phpunit/phpunit": "^10.5.5 || ^11.1.3"
+        "phpstan/phpstan": "1.12.0",
```

Comment:
use latest phpstan 1.12.3 to allow install latest rector 1.2.5, also, why you create separate PR? and not just let me continue my PR there

- https://github.com/cakephp/cakephp/pull/17899

imo, start adding PR should be step by step, you're adding too many rules at once on this PR :)

Reply:
Updating phpstan should not be part of this PR. All the rules that im using are already merged into 5.next so there is no need to make it step by step. All the rules made changes to 5 files.